lib/Platform/Intl/PlatformIntlApple.mm

```diff
@@ -337,48 +490,1338 @@ LocaleMatch lookupMatcher(
+    // This currently outputs to 3 digits only with the date?
+    switch (impl_->FractionalSecondDigits) {
+      case 1: {
+        [customFormattedDate appendString:@"S"];
+      }
+      case 2: {
+        [customFormattedDate appendString:@"SS"];
+      }
+      case 3: {
+        [customFormattedDate appendString:@"SSS"];
+      }
+    }
+  }
+  // Not supported - dayPeriod (at night/in the morning)
+  // Set the custom date from all the concatonated NSStrings (locale will
+  // automatically seperate the order) Only set a template format if it isn't
+  // empty
+  if (![customFormattedDate isEqual:@""]) {
+    [dateFormatter setLocalizedDateFormatFromTemplate:customFormattedDate];
+  }
+
+  NSString *df = dateFormatter.dateFormat;
+  return nsStringToU16String(df);
+}
 std::u16string DateTimeFormat::format(double jsTimeValue) noexcept {
-  auto s = std::to_string(jsTimeValue);
-  return std::u16string(s.begin(), s.end());
+  auto timeInSeconds = jsTimeValue / 1000;
+  NSDate *date = [NSDate dateWithTimeIntervalSince1970:timeInSeconds];
+  NSString *df = u16StringToNSString(getNSDateFormat());
+  NSDateFormatter *dateFormatter = [[NSDateFormatter alloc] init];
+  dateFormatter.dateFormat = df;
+  // Resetting these as it wouldn't pass through getNSDateFormat
+  if (impl_->TimeZone != u"") {
+    dateFormatter.timeZone =
+        [[NSTimeZone alloc] initWithName:u16StringToNSString(impl_->TimeZone)];
+  }
+  if (impl_->locale != u"") {
+    dateFormatter.locale = [[NSLocale alloc]
+        initWithLocaleIdentifier:u16StringToNSString(impl_->locale)];
+  }
+  if (impl_->Calendar != u"") {
+    dateFormatter.calendar = [[NSCalendar alloc]
+        initWithCalendarIdentifier:u16StringToNSString(impl_->Calendar)];
+  }
+  // If no options have been set, date will be empty, so set a format for
+  // default
+  if ([[dateFormatter stringFromDate:date] isEqual:@""]) {
+    dateFormatter.dateStyle = NSDateFormatterShortStyle;
+  }
+  return nsStringToU16String([dateFormatter stringFromDate:date]);
+}
+std::u16string returnTypeOfDate(const char &formatter) {
+  if (formatter == 'a') {
+    return u"dayPeriod";
+  }
+  if (formatter == 'z' || formatter == 'v' || formatter == 'O') {
+    return u"timeZoneName";
+  }
+  if (formatter == 'G') {
+    return u"era";
+  }
+  if (formatter == 'y') {
+    return u"year";
+  }
+  if (formatter == 'M') {
+    return u"month";
+  }
+  if (formatter == 'E') {
+    return u"weekday";
+  }
+  if (formatter == 'd') {
+    return u"day";
+  }
+  if (formatter == 'h' || formatter == 'k' || formatter == 'K' ||
+      formatter == 'H') {
+    return u"hour";
+  }
+  if (formatter == 'm') {
+    return u"minute";
+  }
+  if (formatter == 's') {
+    return u"second";
+  }
+  if (formatter == 'S') {
+    return u"fractionalSecond";
+  }
+  return u"literal";
+}
+std::u16string removeDuplicatesFromString(const std::u16string &s) {
+  std::u16string arr;
+  std::unordered_map<char, char> exists;
```

Comment:
I would be cleaner to use `unordered_set` here.

Reply:
Unfortunately it says no template named 'unordered_set' in namespace 'std';.

Reply:
You have to `#include <unordered_set>`